scripts/cdist

```diff
@@ -101,6 +130,21 @@ def commandline():
         logging.root.setLevel(logging.INFO)
     if args.debug:
         logging.root.setLevel(logging.DEBUG)
+    args_dict = vars(args)
+    # if command with remote_copy and remote_exec params
+    if 'remote_copy' in args_dict and 'remote_exec' in args_dict:
+        # if remote-exec and/or remote-copy args are None then user
+        # didn't specify command line options nor env vars:
+        # inspect multiplexing options for default cdist.REMOTE_COPY/EXEC
+        if args_dict['remote_copy'] is None or args_dict['remote_exec'] is None:
+            control_path_dir = tempfile.mkdtemp(prefix="cdist.control.path")
+            import atexit
+            atexit.register(lambda: shutil.rmtree(control_path_dir))
+            mux_opts = inspect_ssh_mux_opts(control_path_dir)
+            if args_dict['remote_exec'] is None:
+                args.remote_exec = cdist.REMOTE_EXEC + mux_opts
+            if args_dict['remote_copy'] is None:
+                args.remote_copy = cdist.REMOTE_COPY + mux_opts
```

Comment:
I already made the changes. See comments below and code diff.

**All work and no play makes Jack a dull boy**
On Mar 29, 2016 5:38 PM, "Nico Schottelius" <notifications@github.com>
wrote:

> In scripts/cdist
> <https://github.com/ungleich/cdist/pull/418#discussion_r57745497>:
>
> > @@ -101,6 +130,21 @@ def commandline():
> >          logging.root.setLevel(logging.INFO)
> >      if args.debug:
> >          logging.root.setLevel(logging.DEBUG)
> > +    args_dict = vars(args)
> > +    # if command with remote_copy and remote_exec params
> > +    if 'remote_copy' in args_dict and 'remote_exec' in args_dict:
> > +        # if remote-exec and/or remote-copy args are None then user
> > +        # didn't specify command line options nor env vars:
> > +        # inspect multiplexing options for default cdist.REMOTE_COPY/EXEC
> > +        if args_dict['remote_copy'] is None or args_dict['remote_exec'] is None:
> > +            control_path_dir = tempfile.mkdtemp(prefix="cdist.control.path")
> > +            import atexit
> > +            atexit.register(lambda: shutil.rmtree(control_path_dir))
> > +            mux_opts = inspect_ssh_mux_opts(control_path_dir)
> > +            if args_dict['remote_exec'] is None:
> > +                args.remote_exec = cdist.REMOTE_EXEC + mux_opts
> > +            if args_dict['remote_copy'] is None:
> > +                args.remote_copy = cdist.REMOTE_COPY + mux_opts
>
> Sounds good to me!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ungleich/cdist/pull/418/files/4318d72524f4569addf7bd06fd85e74dc5b054df#r57745497>
>